sdk/eventhub/azure-eventhub/azure/eventhub/aio/_error_async.py

```diff
@@ -34,10 +34,10 @@
         name = cast("ClientBaseAsync", closable)._container_id
     if isinstance(exception, KeyboardInterrupt):  # pylint:disable=no-else-raise
         _LOGGER.info("%r stops due to keyboard interrupt", name)
-        await cast("ConsumerProducerMixin", closable).close()
+        await cast("ConsumerProducerMixin", closable)._close_connection_async()
         raise error
     elif isinstance(exception, EventHubError):
-        await cast("ConsumerProducerMixin", closable).close()
+        await cast("ConsumerProducerMixin", closable)._close_connection_async()
```

Comment:
You could combine these two and just log the error:
```
if isninstance(exception, (KeyboardInterrupt, EventHubError)):
    _LOGGER.info("%r shutting down due to %r", name, exception)
    await cast("ConsumerProducerMixin", closable)._close_connection_async()
```
Same comment applies to `exceptions.py` line 167

Reply:
Combined them and further combined with the else statement